article/results.tex

```diff
@@ -56,59 +56,75 @@ \subsection{Best Practice Guidelines}
+\texttt{set -u} would change the default behavior to error out and stop execution as soon as undefined variable is attempted to be used.
+To summarize, we advocate to include \texttt{set -eu} at the top of every shell script to guarantee it to error out as soon as any command fails or an undefined variable is attempted to be used.
+Read more in the section on \href{https://www.repronim.org/module-reproducible-basics/01-shell-basics/\#fail-early}{``Command line/shell: Fail early''} subsection of the ``ReproNim Reproducible Basics Module''~\cite{repronim:reprobasics}.
 
 \subsubsection{Avoid assuming or hard-coding absolute paths to resources}
 Ensuring layout compatibility in different article execution environments is contingent on executable resources being able to find code or data required for execution.
-This would require changes to executable files contingent on higher-level process coordination, which would make article code less portable.
-This problem is best avoided by parameterizing paths to external resources in article code, and allowing it to be specified dynamically or via an environment variable during execution.
+Explicitly hardcoded in the scripts absolute paths are likely to not exist anywhere but the original execution environment, and hence scripts guaranteed to fail.
+This problem is best avoided by adhering to YODA principles~\cite{yoda} of being able to reference all needed resources (data, scripts, container images, etc.) \emph{under} the study folder.
+Use of relative paths within the study scripts then improve their portability.
+Paths to external resources (scratch folders, reusable resources such as atlases etc. if not present within study folder) should be allowed to be specified dynamically via command line options or via an environment variable during execution.
 
 \subsubsection{Avoid assuming a directory context for execution}
-Within the article code, resources may be linked via relative paths, these are paths resolved based on their hierarchical location with the execution base path as the reference point.
-However, if the execution base is variable, as it may be if the environment is modified, these relative paths will become fragile.
-Further, relative paths may also become fragile during debugging, where individual execution from their base directory might be preferable to emulating the otherwise more common base.
-A good way of making scripts utilising relative paths more robust is ensuring that they always set their base execution directory to their parent directory.
-This can be accomplished in Bash via prepending \texttt{cd \textquotedbl\$(dirname \textquotedbl\$0\textquotedbl)\textquotedbl} to the script content.
+Per previous recommendation, resources may be linked via relative paths, these are paths resolved based on their hierarchical location with the execution base path as the reference point.
```

Comment:
I don't think we need hierarchical here, could have a 0-level hierarchy, the point is that they are relative.

Reply:
```suggestion
Per previous recommendation, resources may be linked via relative paths, these are paths resolved based on their  location in the filesystem hierarchy with the execution base path as the reference point.
```
does this make the point I had in mind better?  I think "location" by itself isn't hinting on the path of hierarchical organization of the filesystem

Reply:
yes.